### PR TITLE
[Merged by Bors] - Use plugin setup for resource only used at setup time

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- Build on #6336 for more plugin configurations

## Solution

- `LogSettings`, `ImageSettings` and `DefaultTaskPoolOptions` are now plugins settings rather than resources

---

## Changelog

- `LogSettings` plugin settings have been move to `LogPlugin`, `ImageSettings` to `ImagePlugin` and `DefaultTaskPoolOptions` to `CorePlugin`

## Migration Guide

The `LogSettings` settings have been moved from a resource to `LogPlugin` configuration:

```rust
// Old (Bevy 0.8)
app
  .insert_resource(LogSettings {
    level: Level::DEBUG,
    filter: "wgpu=error,bevy_render=info,bevy_ecs=trace".to_string(),
  })
  .add_plugins(DefaultPlugins)

// New (Bevy 0.9)
app.add_plugins(DefaultPlugins.set(LogPlugin {
    level: Level::DEBUG,
    filter: "wgpu=error,bevy_render=info,bevy_ecs=trace".to_string(),
}))
```

The `ImageSettings` settings have been moved from a resource to `ImagePlugin` configuration:

```rust
// Old (Bevy 0.8)
app
  .insert_resource(ImageSettings::default_nearest())
  .add_plugins(DefaultPlugins)

// New (Bevy 0.9)
app.add_plugins(DefaultPlugins.set(ImagePlugin::default_nearest()))
```

The `DefaultTaskPoolOptions` settings have been moved from a resource to `CorePlugin::task_pool_options`:

```rust
// Old (Bevy 0.8)
app
  .insert_resource(DefaultTaskPoolOptions::with_num_threads(4))
  .add_plugins(DefaultPlugins)

// New (Bevy 0.9)
app.add_plugins(DefaultPlugins.set(CorePlugin {
  task_pool_options: TaskPoolOptions::with_num_threads(4),
}))
```

### CODE REVIEW

File: crates/bevy_log/src/lib.rs [hunk 1]
```diff
@@ -111,6 +103,7 @@ impl Default for LogSettings {
 }
 
 impl Plugin for LogPlugin {
+    #[allow(unused_variables)]
```
Comment (alice-i-cecile) [inline crates/bevy_log/src/lib.rs:106]: Should we just be using `_app` here?

Comment (mockersf) [reply crates/bevy_log/src/lib.rs]: no, `app` is used if feature `tracing-chrome` is enabled...

**Review by JoJoJet — APPROVED** [review]

**Review by IceSentry — APPROVED** [review]

**Comment by DGriffin91** [comment]
Is there a way to query what the default sampler is now? I had code depending on this.

**Comment by mockersf** [comment]
@DGriffin91 would #6372 work for you?

**Comment by DGriffin91** [comment]
@mockersf Yep! Just tested it and it works great! Also got a good laugh when I hovered `get_added_plugins` in vs code and saw the example.

**Pull request closed without merging**